src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,121 @@
+        for (int i = 1; i <= tasks.size(); i++) {
+            response += i + ". " + tasks.get(i - 1) + "\n";
+        }
+        return response;
+    }
+
+
+    /**
+     * finds and prints task based on whether it contains the given keyword
+     *
+     * @param keyword word to be matched with task description
+     */
+    String findTask(String keyword) {
+        String response = "Here are the matching tasks in your list:\n";
+        int currCount = 1;
+        for (int i = 1; i <= tasks.size(); i++) {
+            if (tasks.get(i - 1).getDescription().contains(keyword)) {
+                response += currCount++ + ". " + tasks.get(i - 1) +"\n";
+            }
+
+        }
+        return response;
+    }
+
+
+    /**
+     * marks the chosen task as done and prints a confirmation
+     *
+     * @param taskNumber represents which task to mark, 1 for the first task, 2 for the second, etc
+     */
+    String mark(int taskNumber) {
+        tasks.get(taskNumber - 1).mark();
+        String response = "Nice! I've marked this task as done:\n";
+        response += tasks.get(taskNumber - 1);
+        return response;
+    }
+
+    /**
+     * unmarks the chosen task and prints a confirmation
+     *
+     * @param taskNumber represents which task to unmark, 1 for the first task, 2 for the second, etc
+     */
+    String unmark(int taskNumber) {
+        tasks.get(taskNumber - 1).unmark();
+        String response = "OK, I've marked this task as not done yet:";
+        response += tasks.get(taskNumber - 1);
+        return response;
+    }
+
+    /**
+     * adds the given ToDo object and prints a confirmation
+     *
+     * @param toDo ToDo object to be added to TaskList
+     */
+    String addToDo(ToDo toDo) {
+        tasks.add(toDo);
+        return "Got it. I've added this task:\n"
+                + toDo + "\n"
+                + String.format("Now you have %d tasks in the list."
+                , tasks.size());
+    }
+
+    /**
+     * adds the given Deadline object and prints a confirmation
+     *
+     * @param deadline Deadline object to be added to TaskList
+     */
+    String addDeadline(Deadline deadline) {
+        tasks.add(deadline);
+        return "Got it. I've added this task:\n"
+                + deadline + "\n"
+                + String.format("Now you have %d tasks in the list."
+                , tasks.size());
+    }
+
+    /**
+     * adds the given Event object and prints a confirmation
+     *
+     * @param event Event object to be added to TaskList
+     */
+    String addEvent(Event event) {
+        tasks.add(event);
+        return "Got it. I've added this task:\n"
+                + event + "\n"
+                + String.format("Now you have %d tasks in the list."
+                , tasks.size());
+
+    }
+
+    /**
+     * deletes task at a specified position and prints a confirmation
+     *
+     * @param number represents which task to delete, 1 for the first task, 2 for the second, etc
```

Comment:
I think you have done a thorough job in writing your javadocs; however, just nitpicking here to bring minute details to your attention:) 
"Punctuation behind each parameter description". In that sense, there would be a period after 'etc' and similarly for other javadocs:))

Reply:
Yup, it appears I have missed out on all the full stops after each line, I will fix this issue, thanks for spotting.